### PR TITLE
This adds a simple sample demonstrating Secret Volumes

### PR DESCRIPTION
**Pull request by mattmoor**
This requires this new feature available in 0.4: https://github.com/knative/serving/pull/3061

### CODE REVIEW

**Comment by mattmoor** [comment]
/hold

Until at least the feature has merged :)

File: serving/samples/secrets-go/README.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+   INGRESSGATEWAY=knative-ingressgateway
+
+   # The use of `knative-ingressgateway` is deprecated in Knative v0.3.x.
+   # Use `istio-ingressgateway` instead, since `knative-ingressgateway`
+   # will be removed in Knative v0.4.
+   if kubectl get configmap config-istio -n knative-serving &> /dev/null; then
+       INGRESSGATEWAY=istio-ingressgateway
+   fi
+
+   kubectl get svc $INGRESSGATEWAY --namespace istio-system
```
Comment (vagababov) [inline serving/samples/secrets-go/README.md]: `kubectl get svc $INGRESSGATEWAY --namespace istio-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}'` ?
Or just
EXTERNAL_IP_ADDRESS=$(kubectl get svc $INGRESSGATEWAY --namespace istio-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}')

Comment (mattmoor) [reply serving/samples/secrets-go/README.md]: This is consistent with the `helloworld-go` (and numerous other) docs we have, so I'm going to bias for consistency here.

File: serving/samples/secrets-go/README.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+   ```shell
+   gcloud iam service-accounts create knative-secrets
+   ```
+
+1. Create a new JSON key for this account
+
+   ```shell
+   gcloud iam service-accounts keys create robot.json \
+     --iam-account=knative-secrets@$PROJECT_ID.iam.gserviceaccount.com
```
[the remark this answers is not in the document]
Comment (mattmoor) [reply serving/samples/secrets-go/README.md]: I've updated the wording at the top to **_also_** steal from the place I stole this (gcp-pubsub-source docs)

**Comment by mattmoor** [comment]
@samodell @RichieEscarez What's our policy on merging docs for 0.4?  Wait for 0.4 to cut?

**Comment by samodell** [comment]
@mattmoor  Yes, wait to merge this for now. We're close to having master be the development branch for the next release, but we're not quite there yet.

**Comment by mattmoor** [comment]
/hold cancel

Release branch is cut

**Pull request merged**